app/queries/my_facilities/registrations_query.rb

```diff
@@ -0,0 +1,59 @@
+class MyFacilities::RegistrationsQuery
+  include QuarterHelper
+  include MonthHelper
+  include DayHelper
+
+  attr_reader :periods
+
+  def initialize(period: :quarter, facilities: Facility.all, include_quarters: 3, include_months: 3, include_days: 7)
+    @facilities = facilities
+    @period = period
+    @periods = case period
+               when :quarter then last_n_quarters(n: include_quarters, include_current_quarter: true)
+               when :month then
+                 last_n_months(n: include_months, include_current_month: true)
+                 .map { |month| [month.year, month.month] }
+               when :day then last_n_days(n: include_days)
+               end
+  end
+
+  def registrations
+    case @period
+    when :quarter then quarterly_registrations
+    when :month then monthly_registrations
+    when :day then daily_registrations
+    end
+  end
+
+  def all_time_registrations
+    LatestBloodPressuresPerPatient
+      .where(facility: @facilities)
+      .where('patient_recorded_at < ?', Time.current.beginning_of_day - 2.months)
+  end
+
+  private
+
+  def quarterly_registrations
+    year_quarter_tuples = @periods.map { |(year, quarter)| "('#{year}', '#{quarter}')" }.join(',')
+
+    PatientRegistrationsPerDayPerFacility
+      .where(facility: @facilities)
+      .where("(year, quarter) IN (#{year_quarter_tuples})")
+  end
+
+  def monthly_registrations
+    year_month_tuples = @periods.map { |(year, month)| "('#{year}', '#{month}')" }.join(',')
+
+    PatientRegistrationsPerDayPerFacility
+      .where(facility: @facilities)
+      .where("(year, month) IN (#{year_month_tuples})")
```

Comment:
The other (non-postgres, activerecord/arel) way to say this would be:
`SELECT * WHERE (year = year1 AND month = month1) OR (year = year2 AND month = month2)`

Couldn't figure out a way to generate this sanely with arel/activerecord, mostly because they aren't good at composing OR's. 

```ruby
# Activerecord
periods.reduce(PatientRegistrationsPerDayPerFacility) do
  |query, (year, period)| query.or(PatientRegistrationsPerDayPerFacility.where(year: year).where(@period => period))
end

# Arel
t = PatientRegistrationsPerDayPerFacility.arel_table
periods.reduce(PatientRegistrationsPerDayPerFacility) do |query, (year, a)| 
  query.or(table.where(t[:year].eq(year)).where(t[period].eq(a))) 
end
```
Both of these generate queries that return good data, but the queries themselves look quite horrid.